src/h5web/visualizations/shared/AxisSystemProvider.tsx

```diff
@@ -1,24 +1,45 @@
-import React, { createContext, ReactElement, ReactNode } from 'react';
-import { AxisConfig } from './models';
+import React, { ReactElement, ReactNode } from 'react';
+import { scaleLinear, scaleSymlog } from 'd3-scale';
+import { AxisConfig, AxisInfo } from './models';
+import { isIndexAxisConfig } from './utils';
+import AxisSystemContext from './AxisSystemContext';
 
-export interface DisplayAxisProps {
-  abscissaConfig: AxisConfig;
-  ordinateConfig: AxisConfig;
-}
+function getAxisInfo(config: AxisConfig): AxisInfo {
+  if (isIndexAxisConfig(config)) {
+    const { indexDomain, showGrid = false } = config;
+    return {
+      isIndexAxis: true,
+      scaleFn: scaleLinear,
+      domain: indexDomain,
+      isLog: false,
+      showGrid,
+    };
+  }
 
-export const AxisSystemContext = createContext<DisplayAxisProps>(
-  {} as DisplayAxisProps
-);
+  const { dataDomain, isLog = false, showGrid = false } = config;
+  return {
+    isIndexAxis: false,
+    scaleFn: isLog ? scaleSymlog : scaleLinear,
+    domain: dataDomain,
+    isLog,
+    showGrid,
+  };
+}
```

Comment:
Messy diff, but this `getAxisInfo` function is the core of this refactoring: it takes an `AxisConfig` and returns an `AxisInfo`. This is where I moved the `getDataScaleFn` logic for now, by the way.